### PR TITLE
[MERGE][REF] pos*: bootstrap refactor

### PR DESCRIPTION
**Pull request by vlst-odoo**
This PR contains 2  commits. The first handles the bootstrap refactor and the
second refactors different views. The  changes  brought in the 2 commits are
described below.

**1st commit:**
Prior to this commit, the point_of_sale module relied on custom CSS.

To enhance user experience, increase flexibility, and simplify maintenance,
most of the custom css was replaced in this PR with Bootstrap utility classes.

**2nd commit**
In this commit we refactor some parts of the pos ui with the interest
of providing a simpler and cleaner codebase.

One pattern that is heavily used in the pos ui is that in which
views are changed based on the screen size at the component level.
This means that it becomes very difficult to understand what parts
of a specific template will be seen in mobile or in desktop views.
In most cases, the pos adapts for mobile screens by using the same
main components, but in different configurations. For example, a
component that was visible in desktop mode, might become accesible
through a menu in mobile mode.

In order to improve the situation, the proposed solution is to
separate the logical parts of the ui into `owl templates` and
to inject this templates wherever the need arises.
Thus, instead of constantly using `t-if="ui.isSmall"` in all places
where there is a difference between the mobile and desktop views,
we have a main conditional branch at the root level of the component
representing a certain screen and then build 2 separate views, one for
mobile and one for desktop, each calling the needed templates.

In this commit we implement this pattern for the `payment_screen`.
This will allow us to observe this pattern over a period of time,
before implementing it on the other screens.

In this commit we also:
- add small fixes and improvements relating to
  the previous commit, in which the pos was
  refactored to bootstrap;
- replace the `<br>` tags in the numpad component
  with the bootstrap grid system;
- adapt the `cash_move_popup` to rely more on `t-model`
  and less on `t-on-change`

Task: 3354582

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_restaurant/static/src/app/floor_screen/table.xml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
     <t t-name="pos_restaurant.Table" owl="1">
-        <div class="table" t-on-click.stop="(ev) => props.onClick(props.table, ev)" t-att-style="style">
+        <div class="table position-absolute d-flex flex-column align-items-center justify-content-between rounded cursor-pointer" t-on-click.stop="(ev) => props.onClick(props.table, ev)" t-att-style="style">
             <div class="infos">
-                <span class="label">
+                <span class="label fw-bolder fs-4 z-index-1">
                     <t t-esc="props.table.name" />
                 </span>
-                <span t-att-class="orderCountClass" t-att-hidden="orderCount === 0" t-att-style="badgeStyle">
+                <span t-att-class="orderCountClass" t-attf-class="{{ orderCount === 0 ? 'opacity-0' : ''}}" t-att-style="badgeStyle">
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/app/floor_screen/table.xml:9]: what's the advantage of this change?

Comment (vlst-odoo) [reply addons/pos_restaurant/static/src/app/floor_screen/table.xml]: this was changed back

**Comment by caburj** [comment]
robodoo rebase-merge

**Review by caburj — APPROVED** [review]
robodoo r+

**Pull request closed without merging**